Add paystar gateway

majidfeiz opened the pull request:
Added PayStar driver
Added PayStar config file
Update documentations

omalizadeh commented on composer.json:
```diff
@@ -34,7 +34,7 @@
     ],
     "require": {
-        "php": "^8.1",
+        "php": "^8.0|^8.1",
```
Minimum version for this branch of package is 8.1 and i don't want to decrease it.

omalizadeh replied:
this is not changed @majidfeiz

omalizadeh commented on config/gateway_paystar.php:
```diff
@@ -0,0 +1,25 @@
+     */
+    'main' => [
+        'pin' => '',
+        'secret' => '',
```
where is 'secret' used?

majidfeiz replied:
It is used to generate a sign key

omalizadeh commented on config/gateway_paystar.php:
```diff
@@ -0,0 +1,25 @@
+     *  gateway configurations.
+     */
+    'main' => [
+        'pin' => '',
```
it was not 'pin' in paystar config. it was something like gateway_id?

omalizadeh commented on src/Drivers/Paystar/Paystar.php:
```diff
@@ -0,0 +1,176 @@
+            'callback' => $callback,
+            'mobile' => $mobile,
+            'order_id' => $this->getInvoice()->getInvoiceId(),
+            'description' => $description,
```
where is 'sign' key from documentation?

omalizadeh commented on src/Drivers/Paystar/Paystar.php:
```diff
@@ -0,0 +1,176 @@
+
+        return [
+            'ref_num'=>$this->getInvoice()->getTransactionId(),
+            'amount'=>$this->getInvoice()->getAmount()
```
again no sign

omalizadeh commented on src/Drivers/Paystar/Paystar.php:
```diff
@@ -0,0 +1,176 @@
+
+    protected function getStatusMessage(int|string $statusCode): string
+    {
+        $messages = [
```
it's not like the status table in the documentation!

omalizadeh replied:
still got problem in messages. -1xx codes are NOT from paystar

majidfeiz replied:
It's here :

https://docs.paystar.ir/docs/tutorial-basics/IPGD

omalizadeh commented on src/Drivers/Paystar/Paystar.php:
```diff
@@ -0,0 +1,176 @@
+        return [
+            'Content-Type' => 'application/json',
+            'Accept' => 'application/json',
+            'Authorization'=> 'Bearer '.$this->settings['pin']
```
you can use withToken on Http facade instead of passing header

omalizadeh reviewed (changes requested):
Check paystar doc carefully

majidfeiz commented:
- [X] secret use for generate sign but sign is not required it's optional and now user can use if need.
- [X] changed pin to gateway_id 
- [X] complete status array
- [X] used withToken on Http facade instead of passing header
- [X] changed PHP version to 8.1

omalizadeh commented on config/gateway_paystar.php:
```diff
@@ -0,0 +1,33 @@
+        'gateway_id' => '',
+        'secret_key' => '',
+        'type' => '',
+        'use_sign' => '',
```
if it's bool why you wrote quotes? set a default value

majidfeiz replied:
Some times developers set secret_key value but they don't need to use sign because it's optional

omalizadeh replied:
when they don't need to use sign they set it to false. use_sign is bool, isn't it? i asked why the quotes?

majidfeiz replied:
Done

omalizadeh reviewed (changes requested):
Check comments

majidfeiz commented on config/gateway_paystar.php:
```diff
@@ -0,0 +1,33 @@
+     *  gateway configurations.
+     * gateway_id is required
+     * type is required => direct | pardakht
+     * use_sign is optional => true | false
```
It's not required. They made a mistake in documentation. I called the support team and talked with a backend team in Paystar.

majidfeiz commented on config/gateway_paystar.php:
```diff
@@ -0,0 +1,33 @@
+    /**
+     *  gateway configurations.
+     * gateway_id is required
+     * type is required => direct | pardakht
```
It has two gateways for payments and the API base url is different

https://docs.paystar.ir/docs/tutorial-basics/IPGD
https://docs.paystar.ir/docs/tutorial-basics/IPG

majidfeiz replied:
I don't know but Paystar has 2 gateways for transactions and this driver can support both of them.

omalizadeh commented:
OK, thanks for PR. please fix errors on action (styles) then we can merge it. @majidfeiz

majidfeiz commented:
I fixed the code style, please merge again, @omalizadeh.

omalizadeh commented:
Still not passing style check. use php cs fixer to automatically fix styles and commit it @majidfeiz
